### PR TITLE
Incorporate Julia Putko's changes

### PR DESCRIPTION
**Pull request by hvdosser**
Fixed conflicts in slocum.py and seaexplorer.py and removed Argo submodules.

### CODE REVIEW

File: pyglider/seaexplorer.py [hunk 1]
```diff
@@ -454,4 +454,3 @@ def _parse_sensor_config(filename):
     return active_device_dicts
 
 
-__all__ = ['raw_to_rawnc', 'merge_rawnc', 'raw_to_timeseries']
```
Comment (jklymak) [inline pyglider/seaexplorer.py:457]: revert this

File: pyglider/website.py [hunk 2]
```diff
@@ -11,6 +11,11 @@
 
 import logging
 
+# jpnote addition for argo float calcs 
+import argopandas as argo 
+import pandas as pd
+# end of jp addition
```
[the remark this answers is not in the document]
Comment (jklymak) [reply pyglider/website.py:17]: These all look useful, but I think we need to move out of pyglider.

**Review by jklymak — CHANGES_REQUESTED** [review]
OK, not sure about the pyglider changes.  We need to move the website stuff out of pyglider and into cproofwebsite.  I've not seen the plotting changes.

File: pyglider/slocum.py [hunk 3]
```diff
@@ -639,6 +639,113 @@ def merge_rawnc(indir, outdir, deploymentyaml,
         ds.to_netcdf(outnebd, 'w')
 
 
+    with open(deploymentyaml) as fin:
```
[the remark this answers is not in the document]
Comment (hvdosser) [reply pyglider/slocum.py:642]: Already done.

File: pyglider/slocum.py [hunk 4]
```diff
@@ -753,6 +867,25 @@ def raw_to_timeseries(indir, outdir, deploymentyaml, *,
 
         ds.attrs['deployment_start'] = str(start)
         ds.attrs['deployment_end'] = str(end)
+
+## jpnote: beginning of jp addition ##
```
[the remark this answers is not in the document]
Comment (hvdosser) [reply pyglider/slocum.py:871]: It's been removed.

File: pyglider/seaexplorer.py (hunk 1 above)
Comment (hvdosser) [reply pyglider/seaexplorer.py:457]: Done.

Comment (jklymak) [reply pyglider/seaexplorer.py:457]: OK, but please add the extra line back in and remove the two spare carriage returns.  We don't have linting turned on, but we will ;-). Liniting ensures some coding consistency and it will complain about both of these changes.  

basically a little trick here is `git restore --source=upstream/main pyglider/seaexplorer.py` to restore the old version.  

Please do the same for slocum below.

File: pyglider/website.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jklymak) [reply pyglider/website.py:17]: The argopanda commit is still in this PR...

For website.py, lets copy the version on `upstream/main` into cproofwebsite, and commit it there (maybe under scripts/ or something).  Then make a pr on the website repo for the changes proposed here.  The website can have argopandas as a dependency (not a submodule ;-)) and we can have the map plotting in there etc.  

Probably the same with mapplotting.py. That is far too specific to our sites to be part of pyglider.

Again, we really want pyglider to be of general use, so we should try and move cproof specific routines out if we can.

File: pyglider/plotting.py [hunk 5]
```diff
@@ -223,55 +363,78 @@ def grid_plots(fname, plottingyaml):
...
                                 sharex=True, sharey=True)
         axs = axs.flat
         for n, k in enumerate(keys):
             _log.debug(f'key {k}')
 
             pconf = config['pcolor']['vars'][k]
             _log.debug(pconf)
-            cmap = pconf.get('cmap', 'viridis')
+
+            cmap = plt.cm.get_cmap('viridis') #jp
```
Comment (jklymak) [inline pyglider/plotting.py:377]: This change makes the colormap un-configurable...

File: pyglider/plotting.py [hunk 6]
```diff
@@ -130,10 +267,10 @@ def timeseries_plots(fname, plottingyaml):
                 _log.info('cl %s', k)
                 if config['timeseries'][k] == 'True':
                     ax = axs[n]
-                    locator = mdates.AutoDateLocator(minticks=3, maxticks=7)
-                    formatter = mdates.ConciseDateFormatter(locator)
-                    ax.xaxis.set_major_locator(locator)
-                    ax.xaxis.set_major_formatter(formatter)
+           #         locator = mdates.AutoDateLocator(minticks=3, maxticks=7)
+           #         formatter = mdates.ConciseDateFormatter(locator)
```
Comment (jklymak) [inline pyglider/plotting.py:271]: we probably want to keep this in here?

File: pyglider/plotting.py [hunk 7]
```diff
@@ -223,55 +363,78 @@ def grid_plots(fname, plottingyaml):
...
 
             ax = axs[n]
-            locator = mdates.AutoDateLocator(minticks=3, maxticks=7)
-            formatter = mdates.ConciseDateFormatter(locator)
-            ax.xaxis.set_major_locator(locator)
-            ax.xaxis.set_major_formatter(formatter)
+            ax.yaxis.set_tick_params(labelbottom=True) 
+            ax.xaxis.set_tick_params(labelbottom=True)
+           # locator = mdates.AutoDateLocator() 
+           # formatter = mdates.ConciseDateFormatter(locator)
```
Comment (jklymak) [inline pyglider/plotting.py:385]: as above

File: pyglider/plotting.py [hunk 8]
```diff
@@ -223,55 +363,78 @@ def grid_plots(fname, plottingyaml):
...
             if vmax is not None:
                 max = vmax
             # make time windows:
+            if ax == axs[0]:
+                vmax = 300
+                cmap.set_over("black") 
+                ax.set_ylim([300, 0])
+
+            vmax = 300
+            cmap.set_over("black")
```
Comment (jklymak) [inline pyglider/plotting.py:404]: I'm not a huge fan ov "over" being coloured black...

File: pyglider/plotting.py [hunk 9]
```diff
@@ -223,55 +363,78 @@ def grid_plots(fname, plottingyaml):
...
             if len(ind) > 1:
                 time = ds.time[ind[1:]] + np.diff(ds.time[ind]) / 2
+              #  print('tim1',time)
                 time = np.hstack((time[0] - (time[1]-time[0]) / 2, time))
+             #   print('tim2',time)
                 depth = ds.depth[:-1] - np.diff(ds.depth)
                 depth = np.hstack((depth, depth[-1] + np.diff(ds.depth)[-1]))
                 pc = ax.pcolormesh(time, depth, ds[k][:, ind],
-                    rasterized=True, vmin=min, vmax=max, cmap=cmap, shading='auto')
+                    rasterized=True, vmin=min, vmax=max,cmap=cmap, shading='auto')
```
Comment (jklymak) [inline pyglider/plotting.py:417]: The space will fail flake8

File: pyglider/plotting.py [hunk 10]
```diff
@@ -223,55 +363,78 @@ def grid_plots(fname, plottingyaml):
...
-                fig.colorbar(pc, ax=ax, extend='both', shrink=0.6)
-            ax.set_title(ds[k].attrs['long_name'] + ' [' +
-                     ds[k].attrs['units'] + ']', loc='left', fontsize=9)
+               # fig.colorbar(pc, ax=ax, extend='both', shrink=0.6)  #jpnote commented
+                cbar = fig.colorbar(pc, ax=ax, extend='both', shrink=0.6)
+                cbar.ax.set_ylabel(ds[k].attrs['long_name'] + ' [' +
+                                 ds[k].attrs['units'] + ']', fontsize=9)
+           # ax.set_title(ds[k].attrs['long_name'] + ' [' +
+                   #  ds[k].attrs['units'] + ']', loc='left', fontsize=9) #jpnote commented 
+            ax.set_title(ds[k].attrs['long_name'] + ' over Depth [m]', loc='left', fontsize=9)
```
Comment (jklymak) [inline pyglider/plotting.py:429]: What does "over Depth [m]" mean?

File: pyglider/plotting.py [hunk 11]
```diff
@@ -223,55 +363,78 @@ def grid_plots(fname, plottingyaml):
...
             t1 = ds.time[-1]
             ax.set_xlim([t0, t1])
-            ax.set_ylim([depmax, 0])
-            if n == 0:
-                ax.set_ylabel('DEPTH [m]')
+           # ax.set_ylim([depmax, 0])
+            ax.set_ylim([300, 0]) #jp changed (or should I leave it as dep max? )
+           # if n == 0:
+            ax.set_ylabel('DEPTH [m]')
+         #   ax.set_xlabel('TIME (May - June)')
```
Comment (jklymak) [inline pyglider/plotting.py:437]: I'd revert these changes...

**Review by jklymak — COMMENTED** [review]
For the plotting, I think my take away is to not change the grid plots.   I don't care about the time series plots as they looked pretty bad.

File: pyglider/website.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hvdosser) [reply pyglider/website.py:17]: Ah, yes that was an empty argopandas directory that's now been removed.

For website.py and mapplotting.py, can I finish this pull request and then move them over into cproofwebsite? It would be nice to not be trying to do two things at once, when I'm struggling with git for both of them.

File: pyglider/seaexplorer.py (hunk 1 above)
Comment (hvdosser) [reply pyglider/seaexplorer.py]: Done.

File: pyglider/plotting.py (hunk 5 above)
Comment (hvdosser) [reply pyglider/plotting.py:377]: Reverted to the original line.

File: pyglider/website.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jklymak) [reply pyglider/website.py:17]: I don't want to add mapplotting.py to git history of this repo (though its not a big deal).  Just remove it from the PR, and move the file to a temperorary location.  

We could merge the changes to website.py, but you may as well just do the same thing - move the version you want to use to a different location.  Why?  Because we know `website.py` works as-is.  When we move it over, it is _far_ better to start with a version that we know works, than one that hasn't been tested yet, and then make PRs on that version (i.e. the changes you propose here).  The advantage is that we can always back the changes out.

File: pyglider/plotting.py (hunk 6 above)
Comment (hvdosser) [reply pyglider/plotting.py:271]: Put it back in.

File: pyglider/plotting.py (hunk 7 above)
Comment (hvdosser) [reply pyglider/plotting.py:385]: as above.

File: pyglider/plotting.py (hunk 8 above)
Comment (hvdosser) [reply pyglider/plotting.py:404]: It was to differentiate it from the gray bathymetry. What colour would be better?

File: pyglider/plotting.py (hunk 9 above)
Comment (hvdosser) [reply pyglider/plotting.py:417]: I put the space back.

File: pyglider/plotting.py (hunk 11 above)
Comment (hvdosser) [reply pyglider/plotting.py:437]: This should be for the new plots that Julia made of different variables vs. depth. She was cutting them off at 300m so you could see the near-surface more easily (since the deep stuff doesn't change much.)

File: pyglider/plotting.py (hunk 8 above)
Comment (hvdosser) [reply pyglider/plotting.py:404]: Actually I've just removed it because it was failing the tests.

File: pyglider/website.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hvdosser) [reply pyglider/website.py]: Done.

File: pyglider/plotting.py (hunk 11 above)
Comment (hvdosser) [reply pyglider/plotting.py:435]: Done.

File: pyglider/plotting.py (hunk 6 above)
Comment (hvdosser) [reply pyglider/plotting.py]: Keeping this part of the code causes an out of index error in matplotlib/dates.py in the testing. I've looked at it, but I don't know enough about dates.py to troubleshoot it very effectively. Did you want me to just keep lines 270 and 271 or the entire block of code following those lines (and similar pieces of code elsewhere in plotting.py?

File: pyglider/plotting.py (hunk 8 above)
Comment (jklymak) [reply pyglider/plotting.py]: OK - usually the over cover is just the max color of the colormap?  Maybe you are referring to the NaN behaviour, which will just put an empty square.  That is indeed hard to differentiate from bathymetry depth unless you have a bathy estimate from elsewhere.

File: pyglider/plotting.py (hunk 6 above)
Comment (jklymak) [reply pyglider/plotting.py]: We could probably figure out what is going on.  But I'm confused because this has been woking, so I'm not clear why it is broken now.

Comment (jklymak) [reply pyglider/plotting.py]: Its because you didn't specify enough empty offsets.

_But_... why would you get rid of the offsets?  The dates can often be meaningless with out them

You guys also changed a bunch of things about the plots that were features not bugs.  You labeled all the ticks on each axes, which were explicitly not labeled in the original.  You changed the date formatting, etc.  I put substantial care into making the grid plots the way I want them.  I'm happy to take suggestion on them, but I'll revert the ones here.

**Comment by jklymak** [comment]
I'll merge this, even though its broken and push a new PR to fix

**Comment by jklymak** [comment]
Hi @hvdosser 

So, this was confusing to deal with because I think you just made a PR on a c-proof branch versus your own branch?  I see you have your own fork so the usual workflow would be:

### clone your own fork locally:

```
git clone git@github.com:hvdosser/pyglider
```

If you do 

```
$ git remote -v
origin  git@github.com:hvdosser/pyglider.git (fetch)
origin  git@github.com:hvdosser/pyglider.git (push)
```

you can see that your branch is the remote named *origin*.


### Add upstream remote:

```
git remote add upstream git@github.com:pyglider/pyglider
$ git remote -v
origin  git@github.com:hvdosser/pyglider.git (fetch)
origin  git@github.com:hvdosser/pyglider.git (push)
upstream        git@github.com:c-proof/pyglider.git (fetch)
upstream        git@github.com:c-proof/pyglider.git (push)
```

Now there is a second remote named *upstream*.  You can fetch it, and you will then have the commits in that remote on your machine:

```
git fetch upstream
```

### Make a *placeholder* branch

This is less confusing than using *main* locally as your default branch.  I do

```
$ git checkout -b placeholder
$ git reset --hard upstream/main
```

That makes the *placeholder* branch an exact copy of the *upstream/main* branch

### Make a new branch

```
$ git checkout placeholder
$ git reset --hard upstream/main
$ git checkout -b my-new-branch
```

and edit away.  However, make pull requests early and often.  They are easier to review and check in about.

### Commit some changes

Some code has changed, and you want to save it as a commit:

``` 
$ git commit -a -m "My description of code changes"
```

### Push to GitHub remote

_Do not_ push to upstream, unless you really really have to!  Always push to _origin_!

```
$ git push origin my-new-branch
```

It should give a message like

```
remote: Create a pull request for 'fix-jp-changes' on GitHub by visiting:
remote:      https://github.com/hvdosser/pyglider/pull/new/my-new-branch
```

You can subsequently keep pushing to that branch as you add more commits.

### Rebasing because upstream has changed

If *upstream/main* changes, then GitHub tries to reconcile your changes with the upstream changes.  You can do this locally with 

```
$ git fetch upstream
$ git rebase upstream/main
```

If there are no conflicts, then your changes do not conflict, and you can push up to origin no problem.  However, because the base of the branch has changed from the old *upstream/main* to the new *upstream/main* you need to force push any changes:

```
$ git push origin my-new-branch --force
```

However, sometimes you will have "merge conflicts" because some one else has changed code you are also trying to change, and git can't figure out whose change to apply.  The rebase will list these files with problems.  For each one you need to 

1. open the conflicted file in an editor.  It will have markers like "<<<<<<<<<" in it denoting the conflicted regions.  You need to look at this by hand and decide which change to keep, and of course edit out the markers.  
2. After the file has all its conflicts reconciled, you need to re-add it `$ git add pyglider/naughty_file.py`

When all the files have been reconciled, you need to continue: `$ git rebase --continue`

Git reconciles each commit one by one, so its possible future commits will also have conflicts that also need to be reconciled.  This is a good reason to keep your number of commits to a bare minimum.

### Squashing commits

When developing, I'll often have a bunch of commits, but when it is time to reconcile with *upstream*, or push to GitHub. all these commits become a liability, so I can squash them.  The most failsafe way to do this is to figure out the last commit that you did not author as part of the PR by doing `git log`. 

```
commit 89805d1047677c87f76f41041c00fdadd016984e (HEAD -> fix-jp-changes, origin/fix-jp-changes)
Author: Jody Klymak <jklymak@gmail.com>
Date:   Thu Jun 16 09:32:14 2022 +0200

    Minor cleanups

commit ade50972704f190e7c72113ccfd31bc571dc144e
Author: Jody Klymak <jklymak@gmail.com>
Date:   Thu Jun 16 09:24:03 2022 +0200

    MNT: revert some things in jp plotting

commit cea08ae90af2be90ce237b416404330b10f03b3a
Author: Jody Klymak <jklymak@gmail.com>
Date:   Thu Jun 16 09:13:11 2022 +0200

    MNT: revert changes to grid plots

commit 6fcd7a7de7ac4a26a720eae6091a7c5d42de7cb8 (upstream/main)
Merge: eb46b71 0d3a214
Author: Jody Klymak <jklymak@gmail.com>
Date:   Thu Jun 16 09:18:19 2022 +0200

    Merge pull request #86 from c-proof/jp_dev
    
    Incorporate Julia Putko's changes

commit 0d3a2146450316ec5a4cad3edb24d8b5bb2aaa1b (upstream/jp_dev)
Author: C-PROOF <cproof@valdez.seos.uvic.ca>
Date:   Wed Jun 15 16:10:10 2022 -0700

    Reverting changes in plotting.py
    
    Fixed an issue with axis labels and with the colormap.
```

So here, the last commit that was on GitHub was `6fcd7a7de7`.  To squash these commits, I do 
```
git rebase -i  6fcd7a7de7ac
```
which will do an interactive rebase.  This should open my editor, and then I choose to "squash" or "pick" the commits.   Int his case I'd squash all except `cea08ae90a`, which I'd "pick".  

After you are done this you can force to your PR: `git push origin my-new-feature --force`.  Note that you have to `--force` again because commits are being destroyed on the origin.
